lib/automate.js

```diff
@@ -0,0 +1,119 @@
+var BaseClient = require("./client");
+var extend = require("./extend");
+var util = require("util");
+
+function AutomateClient(settings) {
+	this.server = {
+		host: "www.browserstack.com"
+	};
+	BaseClient.call(this, settings);
+}
+
+util.inherits(AutomateClient, BaseClient);
+
+// public API
+extend(AutomateClient.prototype, {
+	getPlan: function(fn) {
+		this.request({
+			path: this.path("/plan.json")
+		}, fn);
+	},
+
+	getBrowsers: function(fn) {
+		this.request({
+			path: this.path("/browsers.json")
+		}, fn);
+	},
+
+	getProjects: function(fn) {
+		this.requestAutomate({
+			path: this.path("/projects.json")
+		}, fn);
+	},
+
+	getProject: function(id, fn) {
+		this.requestAutomate({
+			path: this.path("/projects/" + id + ".json")
+		}, fn);
+	},
+
+	getBuilds: function(options, fn) {
+		if (typeof options === "function") {
+			fn = options;
+			options = {};
+		}
+
+		this.requestAutomate({
+			path: this.path("/builds.json")
+		}, fn);
+	},
+
+	getSessions: function(buildId, options, fn) {
+		if (typeof fn === "undefined") {
+			fn = options;
+			options = {};
+		}
+		this.requestAutomate({
+			path: this.path("/builds/" + buildId + "/sessions.json")
+		}, fn);
+	},
+
+	getSession: function(id, fn) {
+		this.requestAutomate({
+			path: this.path("/sessions/" + id + ".json")
+		}, fn);
+	},
+
+	updateSession: function(id, options, fn) {
+		var data = JSON.stringify(options);
+		this.request({
+			method: "PUT",
+			path: this.path("/sessions/" + id + "/.json")
+		}, data, fn);
+	}
+
+});
+
+// internal API
+extend(AutomateClient.prototype, {
+	requestAutomate: function(options, data, fn) {
+		if (typeof data === "function") {
+			fn = data;
+			data = null;
+		}
+
+		fn = fn || function() {};
+
+		var self = this;
+		this.request(options, data, function(error, response) {
+			response = self.cleanResult(response);
+			if (typeof response.builds !== "undefined") {
+				response.builds = self.cleanResult(response.builds);
+			}
+			fn(error, response);
```

Comment:
I'm not sure if you've actually tested this. In the case of a request error, the callback to `request()` will be invoked with a single parameter: `error`. Therefore, the `response` parameter will be `undefined` and `typeof response.builds` will throw an error.

On a slightly more pedantic note, in the case of an error, the callback to `requestAutomate()` should only be invoked with a single parameter in the case of an error, while this will result in two parameters, with the second being `undefined` (if this code were reachable in such as condition).